changelogs/unreleased/5752-davinci26-major.md

```diff
@@ -0,0 +1,14 @@
+## Fix bug with algorithm used to sort Envoy regex/prefix path rules
+
+Envoy is greedy matching path routes and as a result order is important. Contour to produce consistent routing tables in the presence of HTTPProxy inclusion and Ingress Conformance sorts the routes before updating Envoy.
+
+This changes fixes the sorting algorithm used for `Prefix` and `Regex` based path matching. Previously the algorithm lexicographically sorted the regex routes instead of sorting them based on the length of the `Prefix`|`Regex`.
```

Comment:
```suggestion
This changes fixes the sorting algorithm used for `Prefix` and `Regex` based path matching. Previously the algorithm lexicographically sorted based on the path match string instead of sorting them based on the length of the `Prefix`|`Regex`.
```

Reply:
Maybe add a sentence spelling out what the change will be: longer prefix/regexes will be sorted first in order to give preference to more specific routes, then lexicographic sorting for things of the same length

Reply:
fixed in a4ef0f6d